src/OrchardCore.Modules/OrchardCore.Lists/Handlers/ContainedPartLocalizationHandler.cs

```diff
@@ -27,8 +27,8 @@ public override async Task LocalizingAsync(LocalizationContentContext context, L
             // todo: remove this check and change the handler to targe t ContainedPart when issue 3890 is fixed
             if (containedPart != null)
             {
-                var list = await _session.QueryIndex<LocalizedContentItemIndex>(x => x.ContentItemId == containedPart.ListContentItemId).FirstOrDefaultAsync();
-                var localizedList = await _session.QueryIndex<LocalizedContentItemIndex>(x => x.LocalizationSet == list.LocalizationSet && x.Culture == context.Culture).FirstOrDefaultAsync();
+                var list = await _session.QueryIndex<LocalizedContentItemIndex>(i => (i.Published || i.Latest) && i.ContentItemId == containedPart.ListContentItemId).FirstOrDefaultAsync();
+                var localizedList = await _session.QueryIndex<LocalizedContentItemIndex>(i => (i.Published || i.Latest) && i.LocalizationSet == list.LocalizationSet && i.Culture == context.Culture).FirstOrDefaultAsync();
```

Comment:
Don't know enough about this handler to know whether it should be including `Latest` as well?

Reply:
Before, even there was no `Latest` column (only the `Published` column), all latest items was indexed

                    // Inthe index provider
                    if (!contentItem.Latest)
                    {
                        return null;
                    }
                    // All latest are indexed (and that have a loc part and non null Set and Culture)

So, before the query was returning all the latest (published or not) that was matching the predicate

But, because now when we removed an item, and only when we removed an item, new records are added with published = latest = false, so we need to filter these new records to return the same ones as before

As done for autoroutes